crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1225,7 +1225,11 @@ pub async fn filter_payment_methods(
 
                     let connector_variant = api_enums::Connector::from_str(connector.as_str())
                         .into_report()
-                        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+                        .change_context(errors::ConnectorError::InvalidConnectorName)
+                        .change_context(errors::ApiErrorResponse::InternalServerError)
```

Comment:
invalid connector bad request?

Reply:
U mean this cannot be internal server error? could u please provide more info